src/var.c

```diff
@@ -250,11 +271,67 @@ static int fi_parse_bool(const char *str_value)
 	return -1;
 }
 
+static void load_conf() {
+	char *line;
+	char *tmp;
+	struct fi_conf_entry *conf;
+
+	dlist_init(&conf_list);
+
+	FILE *fptr = fopen(DEFAULT_CONF_FILE_PATH, "r");
+	if (!fptr)
+		return;
+
+	line = calloc(1, MAX_CONFIG_LINE_LENGTH);
+	if (!line)
+		goto out;
+
+	while (fgets(line, MAX_CONFIG_LINE_LENGTH, fptr) != NULL) {
```

Comment:
You can use getline() and avoid the hard-coded max line length define and separate line calloc.  getline() returns the length of the string, which makes it easy to check for and replace '\n' with '\0'.  The latter is needed to handle config files that don't end in a new line, which would be easy for a user to not add.

Reply:
Getline is not part of c library, it's a POSIX function. it's unavailable on windows.  Should I have a special case for windows or revert back to fgets?

Reply:
sigh... of course it isn't.  I'd revert back to fgets.  You could define getline for windows in the osd files as an option, but I'm not sure that's worth it.